simulator/resourceapplier/resourceapplier_test.go

```diff
@@ -0,0 +1,521 @@
+package resourceapplier
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	dynamicFake "k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/restmapper"
+	scheduling "k8s.io/kubernetes/pkg/apis/scheduling/v1"
+	storage "k8s.io/kubernetes/pkg/apis/storage/v1"
+)
+
+func TestResourceApplier_createPods(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          string
+		podToApply    *corev1.Pod
+		podAfterApply *corev1.Pod
+		wantErr       bool
+	}{
+		{
+			name: "create a Pod",
+			podToApply: &corev1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pod-1",
+					Namespace: "default",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "container-1",
+							Image: "image-1",
+						},
+					},
+				},
+			},
+			podAfterApply: &corev1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pod-1",
+					Namespace: "default",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "container-1",
+							Image: "image-1",
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			client, mapper := prepare()
+			service := New(client, mapper, Options{})
+
+			p, err := runtime.DefaultUnstructuredConverter.ToUnstructured(tt.podToApply)
+			if err != nil {
+				t.Fatalf("failed to convert pod to unstructured: %v", err)
+			}
+			unstructedPod := &unstructured.Unstructured{Object: p}
+			err = service.Create(context.Background(), unstructedPod)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("createPods() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			got, err := client.Resource(corev1.Resource("pods").WithVersion("v1")).Namespace("default").Get(context.Background(), tt.podToApply.Name, metav1.GetOptions{})
```

Comment:
I guess "pods", "v1", "default" should not be hard-coded, should be read from the test case instead.
Ditto in other tests.